File: src/main/java/com/logicalgeekboy/logical_zoom/java_util/impl/TimerUtil.java
package com.logicalgeekboy.logical_zoom.java_util.impl;



public class TimerUtil {
    public long lastMs = System.currentTimeMillis();

    public boolean delay(float f) {
        return (float)(System.currentTimeMillis() - lastMs) > f;
    }



    public float getElapsedTime() {
        return System.currentTimeMillis() - this.lastMs;
    }

    public void reset() {
        this.lastMs = System.currentTimeMillis();
    }
}

File: src/main/kotlin/com/dov/cm/util/TimerUtil.kt
package com.dov.cm.util

class TimerUtil {
    var lastMs: Long = System.currentTimeMillis()

    fun delay(f: Float): Boolean {
        return (System.currentTimeMillis() - lastMs).toFloat() > f
    }


    val elapsedTime: Float
        get() = (System.currentTimeMillis() - this.lastMs).toFloat()

    fun reset() {
        this.lastMs = System.currentTimeMillis()
    }
}
